pkg/config/remoteconfig/messages.go

```diff
@@ -0,0 +1,283 @@
+package remoteconfig
+
+import (
+	"strings"
+	"time"
+
+	"github.com/ddev/ddev/pkg/config/remoteconfig/internal"
+	"github.com/ddev/ddev/pkg/config/remoteconfig/types"
+	"github.com/ddev/ddev/pkg/dockerutil"
+	"github.com/ddev/ddev/pkg/output"
+	"github.com/ddev/ddev/pkg/styles"
+	"github.com/ddev/ddev/pkg/util"
+	"github.com/ddev/ddev/pkg/versionconstants"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+)
+
+type messageTypes struct {
+	messageType types.MessageType
+	messages    []internal.Message
+}
+
+type conditionDefinition struct {
+	name          string
+	description   string
+	conditionFunc func() bool
+}
+
+var conditionDefinitions = map[string]conditionDefinition{}
+
+func init() {
+	AddCondition("Disabled", "Permanently disables the message", func() bool { return false })
+	AddCondition("Colima", "Running on Colima", dockerutil.IsColima)
+	AddCondition("DockerDesktop", "Running on Docker Desktop", dockerutil.IsDockerDesktop)
+	AddCondition("WSL2", "Running on WSL2", dockerutil.IsWSL2)
```

Comment:
@rfay any other condition we should add?

Reply:
I wish we could consider Colima version and Docker Desktop version, but we don't have easy access to them in golang code at this point.

How about Gitpod and Codespaces while we're here, and OSType?